### PR TITLE
HentaiZ Presence

### PR DESCRIPTION
**Pull request by Kinue72**
image
image

### CODE REVIEW

File: websites/H/HentaiZ/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/685u3IG.png",
+  "thumbnail": "https://i.imgur.com/s1A9MVc.png",
+  "color": "#1f1f1f",
+  "tags": [
+    "anime",
+    "hentai",
+    "nsfw"
+  ],
+  "category": "nsfw",
```
Comment (ririxidev) [inline websites/H/HentaiZ/dist/metadata.json:24]: ``nsfw`` is not valid category.

File: websites/H/HentaiZ/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "vi_VN": "HentaiZ - Trang xem hentai anime vietsub online miễn phí chất lượng cao",
+    "en": "HentaiZ - Top #1 Anime Hentai website in Vietnam with high quality"
+  },
+  "url": [
+    "hentaiz.net",
+    "www.hentaiz.net"
+  ],
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/685u3IG.png",
+  "thumbnail": "https://i.imgur.com/s1A9MVc.png",
```
Comment (ririxidev) [inline websites/H/HentaiZ/dist/metadata.json:17]: Blur every image. We don't want to have nsfw image on bg.

File: websites/H/HentaiZ/presence.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  });
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Date.now(),
+    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
```
Comment (ririxidev) [inline websites/H/HentaiZ/presence.ts:16]: deprecated, use it from Presence class ``presence.getTimestamps()``

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @Kinue72!
You should check out thing(s) that I requested to change in my review. Please look at them!

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

File: websites/H/HentaiZ/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "vi_VN": "HentaiZ - Trang xem hentai anime vietsub online miễn phí chất lượng cao",
+    "en": "HentaiZ - Top #1 Anime Hentai website in Vietnam with high quality"
+  },
+  "url": [
+    "hentaiz.net",
+    "www.hentaiz.net"
+  ],
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/685u3IG.png",
+  "thumbnail": "https://i.imgur.com/rLkSLvZ.png",
```
Comment (ririxidev) [inline websites/H/HentaiZ/dist/metadata.json:17]: Nope, like I said, homepage must be on thumbnail and just blur images?

Comment (Kinue72) [reply websites/H/HentaiZ/dist/metadata.json:17]: kk hold on

File: websites/H/HentaiZ/presence.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    } else {
+      const title = document.querySelector(".anime-name>h1");
+      presenceData.details = title ? title.innerHTML : "Đang ở trang chủ";
+      if (title) presenceData.state = "Đang chọn tập";
+    }
+
+    if (!search.startsWith("?s=") && !curPath.includes("/category"))
+      delete presenceData.state;
+    delete presenceData.smallImageKey;
+
```
Comment (Bas950) [inline websites/H/HentaiZ/presence.ts:66]: I dont see smallImageKey set anywhere above, so why delete it?

File: websites/H/HentaiZ/presence.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    presenceData.startTimestamp = browsingStamp;
+    presence.setActivity(presenceData, false);
```
Comment (Bas950) [inline websites/H/HentaiZ/presence.ts:68]: shouldnt need to do startTimestamp again...

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/H/HentaiZ/presence.ts (hunk 6 above)
Comment (ririxidev) [reply websites/H/HentaiZ/presence.ts:67]: It isn't necessary at all?

Comment (Bas950) [reply websites/H/HentaiZ/presence.ts:67]: You already do that on lines +31 to +35

Comment (Kinue72) [reply websites/H/HentaiZ/presence.ts:67]: nope its not working when i remove that line
image

**Review by ririxidev — APPROVED** [review]
yeah ok it's fine

File: websites/H/HentaiZ/presence.ts (hunk 6 above)
Comment (Kinue72) [reply websites/H/HentaiZ/presence.ts:67]: hello can you approve?

**Review by Bas950 — APPROVED** [review]
LGTM

**Pull request merged**